Source/OxyPlot/Rendering/ScreenPoint.cs

```diff
@@ -117,6 +117,63 @@ public static bool IsUndefined(ScreenPoint point)
+        private static bool IsFinite(double f)
+        {
+            return !(double.IsNaN(f) || double.IsInfinity(f));
+        }
+
+        /// <summary>
+        /// Intersects two rays in screen space, determining distance along them where they meet.
+        /// </summary>
+        /// <param name="ray0start">The origin of the first ray.</param>
+        /// <param name="ray0direction">The direction of the first ray.</param>
+        /// <param name="ray1start">The origin of the second ray.</param>
+        /// <param name="ray1direction">The direction of the second ray.</param>
+        /// <param name="lambda0">The distance along the first ray in multiples of its direction.</param>
+        /// <param name="lambda1">The distance along the second ray in multiples of its direction.</param>
+        /// <returns><c>True</c> if the rays intersect, otherwise <c>false</c></returns>
+        public static bool IntersectRayWithRay(ScreenPoint ray0start, ScreenVector ray0direction, ScreenPoint ray1start, ScreenVector ray1direction, out double lambda0, out double lambda1)
+        {
+            var numerator0 = ray1direction.X * (ray0start.Y - ray1start.Y) - ray1direction.Y * (ray0start.X - ray1start.X);
+            var numerator1 = ray0direction.X * (ray1start.Y - ray0start.Y) - ray0direction.Y * (ray1start.X - ray0start.X);
+            var denomonator = ray1direction.X * ray0direction.Y - ray0direction.X * ray1direction.Y;
+
+            lambda0 = numerator0 / denomonator;
+            lambda1 = numerator1 / denomonator;
+
+            return IsFinite(lambda0) && IsFinite(lambda1);
+        }
+
+        /// <summary>
+        /// Intersects two lines in screen space.
+        /// </summary>
+        /// <param name="line0Start">The start point of the first line.</param>
+        /// <param name="line0End">The end point of the first line.</param>
+        /// <param name="line1Start">The start point of the second line.</param>
+        /// <param name="line1End">The end point of the second line.</param>
+        /// <param name="intersection">The point where the two lines intersect.</param>
+        /// <returns><c>True</c> if the lines intersect, otherwise false.</returns>
+        public static bool IntersectLineWithLine(ScreenPoint line0Start, ScreenPoint line0End, ScreenPoint line1Start, ScreenPoint line1End, out ScreenPoint intersection)
+        {
+            var line0Dir = line0End - line0Start;
+            var line1Dir = line1End - line1Start;
+            bool notParallel = IntersectRayWithRay(line0Start, line0Dir, line0Start, line0Dir, out double lambda0, out double lambda1);
+
+            if (!notParallel)
+            {
+                intersection = default;
+                return false;
+            }
+
+            intersection = line1Start + line1Dir * lambda1;
+            return lambda0 >= 0 && lambda0 <= 1 && lambda1 >= 0 && lambda1 <= 1;
+        }
```

Comment:
I don't really like the fact that these are duplicated between `ScreenPoint` and `DataPoint`, but I suppose the same could also be said about other functionality that is common to `ScreenPoint` and `DataPoint`. Maybe we should consider adding something like an `IPoint` interface in the future and implement stuff like this as generic static functions. But I guess this might be out of scope of this PR, so I'm happy if it stays like this for now

Reply:
These statics aren't actually being used; I'm not sure why I added them. I will remove them.